src/karts/kart.cpp

```diff
@@ -508,6 +523,15 @@ void Kart::setPowerup(PowerupManager::PowerupType t, int n)
     m_powerup->set(t, n);
 }   // setPowerup
 
+// ----------------------------------------------------------------------------
+/** Sets the powerup this kart has last used. Number is always 1.
+ *  \param t Type of the powerup.
+ */
+void Kart::setLastUsedPowerup(PowerupManager::PowerupType t)
+{
+    m_last_used_powerup->set(t, 1);
```

Comment:
You appear to use a 'full' Powerup object but only use it to store the TYPE of powerup last used, why? Storing the type might be simpler